### DISCUSSION TITLE
Channel listing

### DISCUSSION THREAD

**Discussion by nanocatdemen** [saleor/saleor#10418] [Q&A] [answered]
Hello!

I need to list channels so the user can select which channel is active on the e-commerce. Since channel listing is private. Is there any way to get them without hardcoding slugs or ids?

**Comment by patrys** [comment, marked as answer]
You could use a privileged call from your server (create a local App to generate an API token with the necessary permissions). We don't want to expose channels by default and there is currently no mechanism to control their visibility.

**Reply by nanocatdemen** [reply]
I see. Thank you for your time and response. I assume the aren't any plans for that. I would gladly add them myself, I would add a Feature request :)

**Reply by nanocatdemen** [reply]
BTW I ended doing something like that but I don't like the request chain call that comes with
